Description: Don't build Files-Excluded
Author: Ximin Luo <infinity0@debian.org>
Forwarded: not-needed
--- a/Makefile.am
+++ b/Makefile.am
@@ -1,9 +1,9 @@
 if BUILD_MICROPY
     MICROPY_DIR = micropython-1.12
 endif
-SUBDIRS =       $(MICROPY_DIR) src check intl po debianold debian doc examples
-## SUBDIRS = 	micropython-1.12 src check intl po debianold debian doc examples
-DIST_SUBDIRS = 	micropython-1.12 src check intl po debianold debian doc examples
+SUBDIRS =       $(MICROPY_DIR) src check intl po doc examples
+## SUBDIRS = 	micropython-1.12 src check intl po doc examples
+DIST_SUBDIRS = 	micropython-1.12 src check intl po doc examples
 EXTRA_DIST = path.h.in setup inst TROUBLES Init.scm compte INSTALACION LEAME.WIN LEAME.ipaq giac.specs giac.specs32 fltk.1.1.changes README.MACOSX pariinl.h changeversion mkosx mkosx10 mkosx3 mkdebian mkdebian32 mkdebian2 win2unix.cc addu.cc xcas*.nsi xcas.desktop xcas.metainfo.xml xcas.xpm xcas.applications README.visualcpp java.README config.h.win config.h.win64 win win64 mkiphone build_for_iphoneos path.iphone simupath.iphone config.h.iphone config.h.64 config.h.mingw path.mingw configure.mingw README.mingw archive.mingw compile.android mkrpm LISEZMOI.numworks icons/* README.hevea2mml README.itex2MML
 
 # Rule to build tar-bzipped distribution package
--- a/doc/el/Makefile.am
+++ b/doc/el/Makefile.am
@@ -1,5 +1,3 @@
-SUBDIRS = tutoriel casinter
-DIST_SUBDIRS = tutoriel casinter
 info_TEXINFOS = giac_el.texinfo
 AM_MAKEINFOHTMLFLAGS = --no-split
 
@@ -15,7 +13,7 @@
 
 the_LANG = el
 thelangdocdir = $(docdir)/$(the_LANG)
-dist_thelangdoc_DATA = html_mall html_mtt html_vall xcasmenu xcasex keywords 
+dist_thelangdoc_DATA = html_mall html_mtt html_vall xcasmenu xcasex keywords
 
 html-local: $(HTML_STAMPS)
 pdf-local: $(PDF_LOCAL)
@@ -76,7 +74,7 @@
 
 .tex.stamp:
 #	$(RUN_LATEX2HTML) $< --address '$(GIAC_ADDRESS)'
-	$(RUN_HEVEA) $< 
+	$(RUN_HEVEA) $<
 	tdir=`echo $< | sed -e 's/\.tex//'`; \
 	imagen -png -pdf -mag 2000 $$tdir; \
 	$(install_sh) -d $$tdir ; \
--- a/doc/Makefile.am
+++ b/doc/Makefile.am
@@ -1,6 +1,6 @@
 SUBDIRS = 	en es fr el local pt zh de
 DIST_SUBDIRS = 	en es fr el local pt zh de 
-XCASHTML = xcasfr.html xcasen.html *.css *.js giacwasm.wasm logo.png giac.tex giacfr.tex test.tex testfr.tex test.html testfr.html algolycee.html graphtheory-user_manual.pdf *.sty simple.html *tableau* magnets html2canvas.js jspdf.umd.min.js khicasnw.html epsilon.dfu apps.tar recovery nws*.html nws.js numworks.js shakeys khi*.bin bootloader.bin
+XCASHTML = xcasfr.html xcasen.html *.css *.js logo.png giac.tex giacfr.tex test.tex testfr.tex test.html testfr.html algolycee.html graphtheory-user_manual.pdf *.sty simple.html *tableau* magnets html2canvas.js jspdf.umd.min.js khicasnw.html epsilon.dfu apps.tar recovery nws*.html nws.js numworks.js shakeys
 EXTRA_DIST = aide_cas README index.html $(XCASHTML) dxcas/* pari/* castex.tex tortue.tex 50-numworks-calculator.rules  # private nwscert.cc -> shakeys
 
 dist_pkgdata_DATA = aide_cas
@@ -9,12 +9,10 @@
 
 install-exec-hook: testfr.html algoseconde.html
 	$(install_sh) -d $(DESTDIR)$(prefix)/share/giac/doc/dxcas
-	$(install_sh) -d $(DESTDIR)$(prefix)/share/giac/doc/pari
 	for dd in $(XCASHTML); do \
 	  $(INSTALL_DATA) $(top_srcdir)/doc/$$dd $(DESTDIR)$(prefix)/share/giac/doc/ ; \
 	done
 	$(INSTALL_DATA) $(top_srcdir)/doc/dxcas/* $(DESTDIR)$(prefix)/share/giac/doc/dxcas
-	$(INSTALL_DATA) $(top_srcdir)/doc/pari/* $(DESTDIR)$(prefix)/share/giac/doc/pari
 	$(INSTALL_DATA) $(top_srcdir)/doc/index.html $(DESTDIR)$(prefix)/share/giac/doc/
 	$(INSTALL_DATA) $(top_srcdir)/doc/README $(DESTDIR)$(prefix)/share/giac/doc/
 	ln -sf ../aide_cas $(DESTDIR)$(prefix)/share/giac/doc/aide_cas
--- a/doc/en/Makefile.am
+++ b/doc/en/Makefile.am
@@ -1,6 +1,6 @@
 TEX_LOCAL = cascmd_en.tex casinter.tex cas.tex slipres.tex giacxcas.tex tutoriel.tex
 
-PDF_LOCAL = cascmd_en.pdf casinter.pdf cas.pdf troussesurvie_en.pdf
+PDF_LOCAL = cascmd_en.pdf casinter.pdf cas.pdf
 
 PS_LOCAL = cascmd_en.ps casinter.ps cas.ps
 
@@ -34,13 +34,12 @@
 $(srcdir)/casinter: casinter.stamp
 $(srcdir)/tutoriel: tutoriel.stamp
 
-install-data-local: $(HTML_STAMPS) troussesurvie_en.pdf # $(PDF_LOCAL) # $(PS_LOCAL) $(DVI_LOCAL) 
-all-local: html-local ps-local dvi-local troussesurvie_en.pdf
+install-data-local: $(HTML_STAMPS) # $(PDF_LOCAL) # $(PS_LOCAL) $(DVI_LOCAL)
+all-local: html-local ps-local dvi-local
 
 install-data-local: all-local
 	$(install_sh) -d $(DESTDIR)$(prefix)/share/giac/doc/$(the_LANG)
 #	$(INSTALL_DATA) $(PDF_LOCAL) $(DESTDIR)$(prefix)/share/giac/doc/$(the_LANG)
-	$(INSTALL_DATA) troussesurvie_en.pdf $(DESTDIR)$(prefix)/share/giac/doc/$(the_LANG)
 #	$(INSTALL_DATA) $(PS_LOCAL) $(DESTDIR)$(prefix)/share/giac/doc/$(the_LANG)
 #	$(INSTALL_DATA) $(DVI_LOCAL) $(DESTDIR)$(prefix)/share/giac/doc/$(the_LANG)
 #	$(INSTALL_DATA) $(HTML_LOCAL) $(DESTDIR)$(prefix)/share/giac/doc/$(the_LANG)
--- a/configure.ac
+++ b/configure.ac
@@ -564,8 +564,6 @@
 doc/es/Makefile
 doc/fr/Makefile
 doc/el/Makefile
-doc/el/tutoriel/Makefile
-doc/el/casinter/Makefile
 doc/local/Makefile
 doc/pt/Makefile
 doc/zh/Makefile
@@ -573,8 +571,6 @@
 intl/Makefile
 po/Makefile.in
 examples/Makefile
-debianold/Makefile
-debian/Makefile
 micropython-1.12/Makefile
 micropython-1.12/docs/Makefile
 micropython-1.12/drivers/Makefile
