Description: Don't run hevea in background
Author: Ximin Luo <infinity0@debian.org>
Forwarded: TODO
--- a/src/icas.cc
+++ b/src/icas.cc
@@ -523,7 +523,7 @@
 	  out.close();
 	  COUT << "File " << outfile << " created" << outfile << endl << "Then I will run pdflatex " << giac::remove_extension(outfile) << endl ;
 	  if (dohevea){
-	    std::string cmd="hevea -fix "+infile_+" &";
+	    std::string cmd="hevea -fix "+infile_;
 	    COUT << "Running " << cmd << endl;
 	    giac::system_no_deprecation(cmd.c_str());
 	  }
@@ -749,8 +749,8 @@
 	if (pos>=0 && pos<ss){
 	  out << s << endl;
 	  out.close();
-	  COUT << "File " << outfile << " created, now running hevea in background and pgiac " << outfile << endl << "Then I will run pdflatex " << giac::remove_extension(outfile) << endl << "For HTML5 output, you can run\nhevea -fix " << giac::remove_extension(infile) << endl;
-	  std::string cmd="hevea -fix "+giac::remove_extension(infile)+" &";
+	  COUT << "File " << outfile << " created, now running hevea and pgiac " << outfile << endl << "Then I will run pdflatex " << giac::remove_extension(outfile) << endl << "For HTML5 output, you can run\nhevea -fix " << giac::remove_extension(infile) << endl;
+	  std::string cmd="hevea -fix "+giac::remove_extension(infile);
 	  giac::system_no_deprecation(cmd.c_str());
 	  cmd=("pgiac "+outfile+" && pdflatex "+giac::remove_extension(outfile)+" && mv "+giac::remove_extension(outfile)+".pdf "+giac::remove_extension(infile)+".pdf");
 	  COUT << cmd << endl;
