From 543c8c87157b6af668f0192cd1d08c42d2b89249 Mon Sep 17 00:00:00 2001
From: Ileana Dumitrescu <ileanadumi95@protonmail.com>
Date: Sat, 5 Nov 2022 14:07:29 +0200
Subject: [PATCH] fix string compiler error

---
 src/Graph.cc | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

--- a/src/Graph.cc
+++ b/src/Graph.cc
@@ -5733,7 +5733,7 @@
   }
 
   bool inputdouble(const char * msg,double & d,GIAC_CONTEXT){
-    const char * ch=fl_input(msg,print_DOUBLE_(d).c_str());
+    const char * ch=fl_input("%s",msg,print_DOUBLE_(d).c_str());
     if (!ch) return false;
     gen g(ch,contextptr);
     g=evalf_double(g,1,contextptr);
