Description: fix compilation with gcc 11
Author: Bernard Parisse and Julien Puydt
Forwarded: written with upstream

--- giac.orig/src/Tableur.cc
+++ giac/src/Tableur.cc
@@ -2404,7 +2404,7 @@
 	}
 	else {
 	  if (inside){
-	    in=apply(in,interval2deuxpoints);
+	    in=giac::apply(in,interval2deuxpoints);
 	    tmp=logisreg?in:symbolic(at_matrix,makevecteur(r2-r1,vc.size(),in));
 	  }
 	  else {
--- giac.orig/src/maple.cc
+++ giac/src/maple.cc
@@ -1044,7 +1044,7 @@
     gen tmp(_exp2pow(_lin(recursive_normal(g,contextptr),contextptr),contextptr));
     vecteur l(lop(tmp,at_arg));
     if (!l.empty()){
-      vecteur lp=*apply(l,gen_feuille)._VECTptr;
+      vecteur lp=*giac::apply(l,gen_feuille)._VECTptr;
       lp=*apply(lp,contextptr,arg_CPLX)._VECTptr;
       tmp=subst(tmp,l,lp,false,contextptr);
     }
@@ -2870,7 +2870,7 @@
     vecteur uv(gen2vecteur(u));
     int uvs=int(uv.size());
     vecteur initcond;
-    aplatir(*apply(initcond0,equal2diff)._VECTptr,initcond);
+    aplatir(*giac::apply(initcond0,equal2diff)._VECTptr,initcond);
     gen f=apply(f0,equal2diff);
     if (n.type!=_IDNT){
       identificateur N(" rsolve_N");
--- giac.orig/src/misc.cc
+++ giac/src/misc.cc
@@ -7287,7 +7287,7 @@
       return gensizeerr(contextptr);
     if (v.size()==1)
       v.push_back(ggb_var(args));
-    gen tmp=apply(v,equal2diff);
+    gen tmp=giac::apply(v,equal2diff);
     vecteur lv=lvarxwithinvsqrt(tmp,v[1],contextptr);
     gen res=lv.size()<2?1:0;
     res.subtype=_INT_BOOLEAN;
--- giac.orig/src/graphe.h
+++ giac/src/graphe.h
@@ -559,7 +559,7 @@
             std::vector<tree_node*> children;
         } tree_node;
         struct kspaths_comparator {
-            bool operator()(const std::pair<gen,tree_node*> &a,const std::pair<gen,tree_node*> &b) {
+            bool operator()(const std::pair<gen,tree_node*> &a,const std::pair<gen,tree_node*> &b) const {
                 if (is_zero(a.first-b.first))
                     return a.second<b.second;
                 return is_strictly_greater(b.first,a.first,context0);
