Description: fix the documentation build on S390X
Author: Bernard Parisse (upstream)
Forwarded: yes

--- giac.orig/src/tex.cc
+++ giac/src/tex.cc
@@ -1320,8 +1320,10 @@
   // assume math mode enabled
   string gen2tex(const gen & e,GIAC_CONTEXT){
     string s;
+#ifndef S390X
     if (has_improved_latex_export(e,s,false,contextptr))
       return s;
+#endif
     switch (e.type){
     case _INT_: case _ZINT: case _REAL:
       if (e.subtype==_INT_BOOLEAN)
