Description: fix issues on 32bits
Author: upstream
Forwarded: will be in next version

--- giac.orig/src/desolve.cc
+++ giac/src/desolve.cc
@@ -114,7 +114,7 @@
     if (!assume_t_in_ab(t,plus_inf,plus_inf,true,true,contextptr))
       return gensizeerr(contextptr);
     gen res=_integrate(makesequence(f*exp(-t*x,contextptr),x),contextptr);
-    if (lop(res,at_integrate).empty())
+    if (lop(res,at_integrate).empty() && lop(res,at_piecewise).empty() && lop(res,at_sign).empty())
       res=-_limit(makesequence(res,x,0,1),contextptr);
     else
       res=undef;
--- giac.orig/src/modpoly.cc
+++ giac/src/modpoly.cc
@@ -15352,7 +15352,10 @@
       return -ZZ2inttype(-z);
     inttype temp(0);
     NTL::ZZ zztemp(z);
-    inZZ2inttype(zztemp,62,temp);
+    if(sizeof(int*) == 4)
+      inZZ2inttype(zztemp,30,temp);
+    else
+      inZZ2inttype(zztemp,62,temp);
     // COUT << "zz2cl_I " << z << " -> " << temp << '\n';
     return temp;
   }
